base/abstractarray.jl

```diff
@@ -323,9 +340,12 @@ function copy!{R,S}(B::AbstractVecOrMat{R}, ir_dest::Range{Int}, jr_dest::Range{
     return B
 end
 
-function copy_transpose!{R,S}(B::AbstractVecOrMat{R}, ir_dest::Range{Int}, jr_dest::Range{Int}, A::AbstractVecOrMat{S}, ir_src::Range{Int}, jr_src::Range{Int})
+function copy_transpose!{R,S}(B::AbstractVecOrMat{R}, ir_dest::Range{Int}, jr_dest::Range{Int},
+                              A::AbstractVecOrMat{S}, ir_src::Range{Int}, jr_src::Range{Int})
     if length(ir_dest) != length(jr_src) || length(jr_dest) != length(ir_src)
-        throw(ArgumentError("source and destination must have same size"))
+        length(ir_dest) != length(jr_src) &&
+            throw(ArgumentError("source ($(length(ir_src))) and destination ($(length(ir_dest))) must have same size"))
+        throw(ArgumentError("source ($(length(jr_src))) and destination ($(length(jr_dest))) must have same size"))
```

Comment:
The only difference from the old code is that it actually shows which dimensions don't match.